src/com/google/enterprise/adaptor/DocumentContentTransformerPipeline.java

```diff
@@ -0,0 +1,70 @@
+package com.google.enterprise.adaptor;
+
+import com.google.common.base.Strings;
+import com.google.common.collect.Lists;
+
+import java.io.OutputStream;
+import java.lang.reflect.Constructor;
+import java.util.List;
+import java.util.Map;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+/**
+ * The document content transformer pipeline holds all document content transformers
+ * and puts them in series connection.
+ *
+ * @author Dominik Weidenfeld (dominik.weidenfeld@twt.de)
+ */
+public class DocumentContentTransformerPipeline {
+
+  private static final Logger LOG =
+      Logger.getLogger(DocumentContentTransformerPipeline.class.getName());
+
+  private List<Map<String, String>> contentTransformers;
+
+  public DocumentContentTransformerPipeline(final List<Map<String, String>> contentTransformers) {
+    this.contentTransformers = contentTransformers;
+  }
+
+  /**
+   * Creates a new transformer pipeline.
+   *
+   * @param original    original content stream
+   * @param contentType content type
+   * @param metadata    metadata
+   * @return stream pipeline or original output stream
+   */
+  public final OutputStream createPipeline(final OutputStream original,
+                                           final String contentType,
+                                           final Metadata metadata) {
+    if (contentTransformers.size() <= 0) {
+      return original;
+    }
+    DocumentContentTransformer contentTransformer = null;
+    for (final Map<String, String> tConfig : contentTransformers) {
+      final String className = tConfig.get("class");
+      if (Strings.isNullOrEmpty(className)) {
+        LOG.log(Level.WARNING, "Document Content Transformer class is missing");
+        continue;
+      }
+      try {
+        //noinspection unchecked
+        final Class<DocumentContentTransformer> clazz =
+            (Class<DocumentContentTransformer>) Class.forName(className);
+        final Constructor<DocumentContentTransformer> constructor =
+            clazz.getConstructor(Map.class, OutputStream.class, String.class, Metadata.class);
+        if (null == contentTransformer) {
+          contentTransformer =
+              constructor.newInstance(tConfig, original, contentType, metadata);
+        } else {
+          contentTransformer =
+              constructor.newInstance(tConfig, contentTransformer, contentType, metadata);
+        }
```

Comment:
Yes, you're right. I changed to order like in this example:

You have 3 Transformers configured (T1, T2, T3).
Now the object will be created like T1(T2(T3(originalStream))).
So if the adaptor pushes content to the stream, the first instance is T1, which calls T2, which calls T3, which calls the original stream.